src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -50,6 +51,29 @@ public function process(ContainerBuilder $container)
     }
 
     /**
+     * Creates a resource to help know if this service has changed.
+     *
+     * @param \ReflectionClass $reflectionClass
+     *
+     * @return AutowireServiceResource
+     */
+    public static function createResourceForClass(\ReflectionClass $reflectionClass)
+    {
+        $constructor = $reflectionClass->getConstructor();
+        $constructorParams = $constructor ? $constructor->getParameters() : array();
+        $constructorArgumentsForResource = array();
+        foreach ($constructorParams as $parameter) {
+            $constructorArgumentsForResource[] = $parameter->getClass();
```

Comment:
and this misses several cases which impact the autowiring:

- default options for args
- classes which don't exist (but autowiring working because the typehint allows injecting ``null``). This would trigger a ReflectionException in your code here, thus breaking autowiring
- setters (as autowiring supports them now IIRC)